client/src/services/UserService.js

```diff
@@ -168,6 +168,35 @@ export class UserService {
 	setDefaultWallet = async (index) => {
 		this.currentWalletIndex = parseInt(index, 10);
 	}
+
+	generateWallets = async (total) => {
+		const user = this.instance;
+		const existingWallets = user.getHDWallet()._derivedWallets || [];
+		if (existingWallets.length >= 10) {
```

Comment:
Consider having a constant called `const MAXIMUM_ALLOCATED_WALLET = 10` in `@cd/helpers/wallet`, then import into wherever it needs.

Reply:
Updated check with `total`